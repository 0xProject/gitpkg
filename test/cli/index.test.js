import cli from '../../src/cli';

describe('while using the gitpkg cli', () => {
  let publishCommand = null;

  beforeAll(async () => {
    publishCommand = (await import('../../src/cli/commands/publish')).handler;
    jest.mock('../../src/cli/commands/publish');
    publishCommand.mockImplementation(() => {});
  });

  afterAll(() => {
    jest.unmock('../../src/cli/commands/publish');
  });

  beforeEach(() => {
    publishCommand.mockReset();
  });

  it('should correctly handle publish command', () => {
    cli(['publish']);
    expect(publishCommand.mock.calls.length).toEqual(1);

    expect(cleanArgs(publishCommand.mock.calls[0][0])).toMatchSnapshot();
  });

  it('should correctly handle publish command with registry arg', () => {
    cli(['publish', '--registry', 'git@github.com:ramasilveyra/public-registry.git']);
    expect(publishCommand.mock.calls.length).toEqual(1);
    expect(cleanArgs(publishCommand.mock.calls[0][0])).toMatchSnapshot();
  });

  it('should correctly handle unknow commands', () => {
    cli(['doesnt-exist']);
    expect(publishCommand.mock.calls.length).toEqual(0);
  });
});

function cleanArgs(args) {
  delete args.$0; // eslint-disable-line no-param-reassign
  return args;
}
