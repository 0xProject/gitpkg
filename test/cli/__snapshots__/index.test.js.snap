// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while using the gitpkg cli should correctly handle publish command 1`] = `
Object {
  "_": Array [
    "publish",
  ],
  "h": false,
  "help": false,
  "r": undefined,
  "v": false,
  "version": false,
}
`;

exports[`while using the gitpkg cli should correctly handle publish command with registry arg 1`] = `
Object {
  "_": Array [
    "publish",
  ],
  "h": false,
  "help": false,
  "r": "git@github.com:ramasilveyra/public-registry.git",
  "registry": "git@github.com:ramasilveyra/public-registry.git",
  "v": false,
  "version": false,
}
`;
